src/backend/InvenTree/users/serializers.py

```diff
@@ -237,19 +248,63 @@ class Meta:
     )
 
 
+class GroupSerializer(InvenTreeModelSerializer):
+    """Serializer for a 'Group'."""
+
+    class Meta:
+        """Metaclass defines serializer fields."""
+
+        model = Group
+        fields = ['pk', 'name', 'permissions', 'roles', 'users']
+
+    def __init__(self, *args, **kwargs):
+        """Initialize this serializer with extra fields as required."""
+        role_detail = kwargs.pop('role_detail', False)
+        user_detail = kwargs.pop('user_detail', False)
```

Comment:
this should probably be a filter

Reply:
That's a good idea. Probably as a separate PR as we have lots of "invisible" filters like this which mutate the API. It needs an overhaul

Reply:
https://github.com/inventree/InvenTree/issues/9481